### ISSUE TITLE
gnu-utility does not work with autosuggestions

### ISSUE THREAD

**Issue by Mr-Tao** [sorin-ionescu/prezto#1382]
With both _gnu-utility_ and _autosuggestions_ enabled autocompletion does not work. Each keypress is accompanied with 
>g[: invalid integer ‘’

message. It works fine without gnu-utility. Result is the same if _autosuggestions_ from prezto is used instead of upstream.

I have reduced my _zplug’s_ **packages.zsh** to:
>zplug "zsh-users/zsh-autosuggestions", defer:3
zplug "modules/gnu-utility", from:prezto, defer:0

`zplug load --verbose`
```
 Load "/usr/local/opt/zplug/repos/sorin-ionescu/prezto/modules/gnu-utility/init.zsh" (modules/gnu-utility)
[zplug] Run compinit
 Load "/usr/local/opt/zplug/repos/zsh-users/zsh-autosuggestions/zsh-autosuggestions.plugin.zsh" (zsh-users/zsh-autosuggestions)
```

**Comment by belak** [comment]
Is there any info about line numbers which came from the error message? Also, please provide the version info which was requested in the issue template (zsh version as well as os version).

**Comment by Mr-Tao** [comment]
@belak The error message I’ve provided is complete and was emitted on every keypress in apparently futile attempt to build the autosuggestion prompt.
```
$ zsh —version
  zsh 5.2 (x86_64-apple-darwin16.0)
$ system_profiler SPSoftwareDataType
  ⋮
  macOS 10.12.5
```

**Comment by belak** [comment]
It looks like it's because we rebind the [ to point to g[ which overrides the builtin. I'm playing with a PR in zsh-autosuggestions to remove usage of [, but I'm running into some other issues.

Alternatively, you could try removing [ from the giant list at the top of the gnu-utility module. If that fixes it, I may just remove it and add a comment because I'm not a fan of shadowing builtins.

**Comment by belak** [comment]
This was fixed in https://github.com/zsh-users/zsh-autosuggestions/commit/940e10a691b34f0281d09794135e8ab81e8fae32 but I generally only update submodules to releases, so the actual fix will have to wait for that.

**Comment by belak** [comment]
This should be resolved as of the autosuggestions v0.4.1 release which was updated in prezto as of 65bcbae. Thanks for reporting this!
